### ISSUE TITLE
Profile Image

### ISSUE THREAD

**Issue by tomchavakis** [taigaio/taiga-front#409]
When you want to change your profile image and the installation is not connected to the internet, doesn't upload the image. 
It would be nice if the gravatar is an option , otherwise not connected to the internet has image problems.

**Comment by superalex** [comment]
Hello @tomchavakis 

Gravatar is actually an option, if the user uploads an image it should be working even if the instance isn't connected to the internet.  Are your MEDIA settings for taiga-back correctly configured?, could you check the url Taiga is trying to use for your uploaded avatar?

**Comment by tomchavakis** [comment]
Although that i have put the static ip at configuration files the image url is http://localhost:8000/media/user/ ........png

**Comment by superalex** [comment]
So that url is the problem, did you overwrite the MEDIA_URL setting?

**Comment by tomchavakis** [comment]
i follow the installation using taiga-scripts , The nginx config has 
location /media {
        alias /home/..../taiga-back/media;
}

and the taiga-back local.py config has

MEDIA_URL = "/media/"

Regards

**Comment by superalex** [comment]
And what url are you using for accessing the web app?

**Comment by natejgardner** [comment]
I'm experiencing the same bug. Was there ever a resolution?

**Comment by superalex** [comment]
Hello @natejgardner what's exactly your situation?. Are you trying to use gravatar or uploading an image to the server doesn't show the correct url?

**Comment by natejgardner** [comment]
To clarify, after deploying taiga with the scripts, neither profile photos nor gravatars are found.

**Comment by superalex** [comment]
Could you show us an url of the avatar taiga is trying to show?

**Comment by natejgardner** [comment]
http://localhost:8000/media/user/4/5/e/d/[long hash here]/solarized.jpg

**Comment by superalex** [comment]
And what url should be?, which one are you using to access Taiga?.

If you need http://localhost:8000 to be another thing you can update your MEDIA_URL and SITES["api"]["domain"] settings.

**Comment by Lc0rE** [comment]
Hello, I'm also encountering this issue. I did install Taiga using scripts and all is working fine except the avatars. If I disable gravatar in the `config.json`, I can see the default profile images working.
When I try to upload and image, the url assigned is something like:

`http://localhost:8000/user/b/7/4/0/96aa7bcf81dae1bfde968f95fde63e33bed50382c6b66dc2add5a73f5abc/download.png.80x80_q85_crop.png`

My taiga-back settings in `local.py`:

```
MEDIA_URL = "[myIP]:8000/media/"
STATIC_URL = "/static/"

# This should change if you want generate urls in emails
# for external dns.
SITES["front"]["domain"] = "localhost:8000"

DEBUG = True
PUBLIC_REGISTER_ENABLED = True

DEFAULT_FROM_EMAIL = "no-reply@example.com"
SERVER_EMAIL = DEFAULT_FROM_EMAIL

#EMAIL_BACKEND = "django.core.mail.backends.smtp.EmailBackend"
#EMAIL_USE_TLS = False
#EMAIL_HOST = "localhost"
#EMAIL_HOST_USER = ""
#EMAIL_HOST_PASSWORD = ""
#EMAIL_PORT = 25
```
where `myIP` contains the static IP address assigned to this server.
I also performed a `circusctl restart` after editing the local.py settings.

Am I doing something wrong?

**Comment by bameda** [comment]
Yes

```python
MEDIA_URL = "[myIP]:8000/media/"
STATIC_URL = "/static/"
```

This two vars should be two urls, the first one should be the url when you are serving the media files

```python
SITES["front"]["domain"] = "localhost:8000"
```

This should be the url of taiga-front

**Comment by natejgardner** [comment]
I made the above changes, but it's still trying to use localhost. I restarted circusd but it didn't solve this.

**Comment by bameda** [comment]
That's the solution, :-(
Try to clean the browser cache.

**Comment by natejgardner** [comment]
I did that, but no luck. Where is taiga-front located on the server? How do permissions get managed for that directory to expose them to the web? Is this in `/var/www/`?

**Comment by bameda** [comment]
In  /home/taiga/taiga-front-dist/ ?

**Comment by arqeco** [comment]
I had the same problem. Uploaded profile images not appearing. The image link points to "localhost:8000" and not to the server own IP (or domain name). I have installed Taiga using setup-server.sh script.

The solution was to change the file ~/taiga-back/settings/local.py according to these instructions: https://github.com/taigaio/taiga-front/issues/251#issuecomment-70216972

```
MEDIA_URL = "http://192.168.0.33:8000/media/"
STATIC_URL = "http://192.168.0.33:8000/static/"
ADMIN_MEDIA_PREFIX = "/static/admin/"
SITES["front"]["domain"] = "192.168.0.33:8000"
```

**Comment by arqeco** [comment]
By the way, uploaded images are fine now, but gravatar images are not working yet.

Link of one gravatar image on Taiga running on my local server:
https://www.gravatar.com/avatar/64e1b8d34f425d19e1ee2ea7236d3028?d=http%3A%2F%2F192.168.0.120%3A8000%2Fv-1489511410824%2Fimages%2Fuser-avatars%2Fuser-avatar-02.png

This above link, when opened on a browser, redirects to this 404 error page:
http://i2.wp.com/192.168.0.120/v-1489511410824/images/user-avatars/user-avatar-02.png
404 Not Found
nginx

**Comment by ghost271994** [comment]
I have installed taiga instance on our environment which is not accessible from outside. After upgrading to 3.0 version am getting 404 error while trying to fetch default user avatar. It should get the default avatar from **http://mydomainname/v-1522826679902/images/user-avatars/user-avatar-01.png** instead of it is reaching out to i0.wp.com for user avatar.
**http://i0.wp.com/mydomainname/v-1522826679902/images/user-avatars/user-avatar-01.png.** 
Is there any workaround so it will get default image from the local system instead of i0.wp.com. I have set gravatar parameter to false.
